app/assets/javascripts/components/Switch.tsx

```diff
@@ -0,0 +1,48 @@
+import { ComponentChildren, FunctionalComponent } from 'preact';
+import { useState } from 'preact/hooks';
+import { HTMLProps } from 'react';
+import {
+  CustomCheckboxContainer,
+  CustomCheckboxInput,
+  CustomCheckboxInputProps,
+} from '@reach/checkbox';
+import '@reach/checkbox/styles.css';
+
+export type SwitchProps = HTMLProps<HTMLInputElement> & {
+  checked?: boolean;
+  onChange: (checked: boolean) => void;
+  children: ComponentChildren;
+};
+
+export const Switch: FunctionalComponent<SwitchProps> = (
+  props: SwitchProps
+) => {
+  const [checkedState, setChecked] = useState(props.checked || false);
```

Comment:
Maybe we can set `props.checked`'s default value here, what do you think about that approach? @baptiste-grob 
```
Switch.defaultProps = {
  checked: false,
}
```

Reply:
I dislike using defaultProps because they break the functional paradigm. Does that even work on functional components actually?

Reply:
Also, same thing that I mentioned on another comment, the name of the function should be `setCheckedState` @baptiste-grob  😄

Reply:
This also doesn't account for using Switch as an uncontrolled component, see my comment below

Reply:
> Does that even work on functional components actually

Yes! :)